debug/org.eclipse.debug.core/core/org/eclipse/debug/core/LaunchAttributeRead.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2025 ArSysOp.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Alexander Fedorov (ArSysOp) - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.debug.core;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Status;
+
+/**
+ *
+ * Reads the value of launch attribute
+ *
+ * @since 3.23
+ */
+public final class LaunchAttributeRead<V> {
+
+	private final ILaunchConfiguration configuration;
+	private final String id;
+	private final Class<V> type;
+	private final Supplier<V> value;
+
+	public LaunchAttributeRead(ILaunchConfiguration configuration, LaunchAttributeDefined<V> defined) {
+		this(configuration, defined.identity().id(), defined.metadata().valueClass(), defined.metadata()::defaultValue);
+	}
+
+	public LaunchAttributeRead(ILaunchConfiguration configuration, String id, Class<V> type, Supplier<V> value) {
+		this.configuration = Objects.requireNonNull(configuration);
+		this.id = Objects.requireNonNull(id);
+		this.type = Objects.requireNonNull(type);
+		this.value = Objects.requireNonNull(value);
+	}
+
+	public V get() throws CoreException {
+		if (String.class.equals(type)) {
+			return type.cast(configuration.getAttribute(id, String.class.cast(value.get())));
+		}
```

Comment:
Couldn't we have this if-chain in the constructor and then encapsulate the action into a lambda and just execute it here?

Reply:
not so sure